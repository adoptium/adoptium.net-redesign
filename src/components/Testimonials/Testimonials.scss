@use "../../styles/variables.scss" as *;

.arrow-button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  overflow: hidden;
}

.arrow-button-progress::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  border: 3px solid transparent;
  border-top-color: $pink;
  animation: spin 8s linear infinite;
}

.slide-in-left {
  animation: slideInLeft 0.5s ease-out forwards;
}

.slide-in-right {
  animation: slideInRight 0.5s ease-out forwards;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@keyframes slideInLeft {
  from {
    transform: translateX(-100%);
    opacity: 0;
  }

  to {
    transform: translateX(0);
    opacity: 1;
  }
}

@keyframes slideInRight {
  from {
    transform: translateX(100%);
    opacity: 0;
  }

  to {
    transform: translateX(0);
    opacity: 1;
  }
}
