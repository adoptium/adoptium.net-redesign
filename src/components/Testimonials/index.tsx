import React, { useState, useEffect } from "react"
import { FaArrowLeft, FaArrowRight } from "react-icons/fa6"
import { AlibabaCloudSlider, MicrosoftSlider } from "./Icons"

import "./Testimonials.scss"

const testimonialData = [
  {
    quote: "Microsoft is proud to be a Strategic Member of the Eclipse Adoptium Working Group, collaborating closely with industry partners to contribute to and benefit from a vibrant open-source ecosystem. Together, we are dedicated to delivering high-quality, secure, and performant Java runtimes that empower our customers and the developer community.",
    name: "Martijn Verburg",
    role: "Principal Group Manager - Java & Go @ Microsoft",
    image: "/images/authors/martijnverburg.jpg",
  },
  {
    quote: "As a member of the Adoptium Working Group, Alibaba Cloud is committed to fostering a vendor-neutral environment that drives the growth of Java Runtime on our global cloud platform. We contribute Alibaba Dragonwell to the Adoptium marketplace, a trusted source for production-ready OpenJDK builds, and actively participate in the EMT4J sub-project within Adoptium to help developers seamlessly upgrade their applications.",
    name: "Sanhong Li",
    role: "Director of Compiler & Runtime, Alibaba Cloud Intelligence",
    image: "/images/quotes/sanhongli.jpg",
  },
  {
    quote: "Azul has been a member of the Eclipse Adoptium Working Group since its formation. As a provider of free and supported OpenJDK builds—including Azul Platform Core (Zulu) and Azul Platform Prime (Zing)—Azul is a passionate contributor and steward for open-source Java. The company is committed to enhancing users’ ability to access and utilize high-quality Java runtimes effectively, while also providing real-time, actionable Java intelligence such as JVM inventory, vulnerability detection, and identification of dead and unused code.",
    name: "Simon Ritter",
    role: "Deputy CTO at Azul Systems",
    image: "/images/quotes/simonritter.jpeg",
  }
]

const Testimonials = () => {
  const [currentTestimonial, setCurrentTestimonial] = useState(0)
  const [slideDirection, setSlideDirection] = useState("right")

  useEffect(() => {
    const timer = setTimeout(() => {
      setCurrentTestimonial((currentTestimonial + 1) % testimonialData.length)
    }, 8000) // Changes every 8 seconds

    return () => clearTimeout(timer)
  }, [currentTestimonial])

  const nextTestimonial = () => {
    setCurrentTestimonial((currentTestimonial + 1) % testimonialData.length)
    setSlideDirection("right")
  }

  const prevTestimonial = () => {
    setCurrentTestimonial(
      (currentTestimonial - 1 + testimonialData.length) %
        testimonialData.length,
    )
    setSlideDirection("left")
  }

  const testimonial = testimonialData[currentTestimonial]
  const testimonialClassName =
    slideDirection === "right" ? "slide-in-right" : "slide-in-left"
  const testimonialKey = `${currentTestimonial}-${slideDirection}`
  const logoSliderChangeHandler = value => {
    if (value === currentTestimonial) {
      return
    } else {
      setCurrentTestimonial(value)
      if (currentTestimonial > value) {
        setSlideDirection("left")
      } else {
        setSlideDirection("right")
      }
    }
  }

  return (
    <section className="bg-[#0E002A] overflow-x-hidden border-t border-[#3E3355]">
      <div className="max-w-screen-xl px-4 py-8 mx-auto text-center lg:py-16 lg:px-6">
        <div className="flex justify-center items-center">
        <figure className="max-w-screen-md mx-auto" key={testimonialKey}>
          <svg
            className="h-12 mx-auto mb-3 text-pink text-pink-400"
            viewBox="0 0 24 27"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.017 18L14.017 10.609C14.017 4.905 17.748 1.039 23 0L23.995 2.151C21.563 3.068 20 5.789 20 8H24V18H14.017ZM0 18V10.609C0 4.905 3.748 1.038 9 0L9.996 2.151C7.563 3.068 6 5.789 6 8H9.983L9.983 18L0 18Z"
              fill="currentColor"
            />
          </svg>
          <blockquote
            className={testimonialClassName}
            style={{ minHeight: "204px" }}
          >
            <span className="md:text-[26px] text-[20px] leading-[34px] md:leading-10 font-semibold  text-white">
              {testimonial.quote}
            </span>
          </blockquote>
          <figcaption className="flex items-center justify-center md:mt-6 mt-0 space-x-3">
            <img
              className="w-6 h-6 mb-0 rounded-full hidden sm:block"
              src={testimonial.image}
              alt={testimonial.name}
            />
            <div className="flex items-center divide-x-2 divide-gray-500 dark:divide-gray-700">
              <div className="px-3 text-pink font-medium text-pink-900">
                {testimonial.name}
              </div>
              <div className="pl-3 text-sm font-light text-gray-500 dark:text-gray-400">
                {testimonial.role}
              </div>
            </div>
          </figcaption>
        </figure>
        </div>
        {/* Navigation Arrows */}
        <div className="flex justify-center items-center max-w-screen-md mx-auto pt-8">
          <button
            className="arrow-button p-2"
            onClick={prevTestimonial}
            aria-label="Previous Testimonial"
          >
            <FaArrowLeft size={20} strokeWidth={1} />
          </button>
          <button
            className="arrow-button arrow-button-progress p-2"
            onClick={nextTestimonial}
            aria-label="Next Testimonial"
          >
            <FaArrowRight size={20} strokeWidth={1} />
            <div className="progress-border"></div>
          </button>
        </div>
        <div className="max-w-[896px] mx-auto flex flex-wrap justify-center items-center mt-10 gap-6 md:space-x-16">
          <button
            onClick={() => logoSliderChangeHandler(0)}
            aria-label="Microsoft Testimonial"
          >
            <MicrosoftSlider current={currentTestimonial} />
          </button>
          <button
            onClick={() => logoSliderChangeHandler(1)}
            aria-label="Aana Testimonial"
          >
            <AlibabaCloudSlider current={currentTestimonial} />
          </button>
        </div>
      </div>
    </section>
  )
}

export default Testimonials
