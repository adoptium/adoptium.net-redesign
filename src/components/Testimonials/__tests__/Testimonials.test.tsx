import React from "react"
import { render, screen, fireEvent, waitFor, act } from "@testing-library/react"
import { describe, expect, it, vi, beforeEach, afterEach } from "vitest"
import Testimonials from "../index"

describe("Testimonials component", () => {
  beforeEach(() => {
    vi.useFakeTimers()
  })

  afterEach(() => {
    vi.restoreAllMocks()
    vi.useRealTimers()
  })

  it("should render correctly", () => {
    const { container } = render(<Testimonials />)
    expect(container).toMatchSnapshot()
  })

  it("displays the first testimonial initially", () => {
    render(<Testimonials />)
    
    // Check if the first testimonial content is displayed
    expect(screen.getByText(/Microsoft is proud to be a Strategic Member/)).toBeTruthy()
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    expect(screen.getByText("Principal Group Manager - Java & Go @ Microsoft")).toBeTruthy()
  })

  it("displays navigation controls", () => {
    render(<Testimonials />)
    
    // Check for navigation buttons
    expect(screen.getByLabelText("Previous Testimonial")).toBeTruthy()
    expect(screen.getByLabelText("Next Testimonial")).toBeTruthy()
    
    // Check for progress indicators
    expect(screen.getByLabelText("Go to testimonial 1")).toBeTruthy()
    expect(screen.getByLabelText("Go to testimonial 2")).toBeTruthy()
    expect(screen.getByLabelText("Go to testimonial 3")).toBeTruthy()
  })

  it("navigates to next testimonial when next button is clicked", () => {
    render(<Testimonials />)
    
    // Initial state - Microsoft testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Click next button
    const nextButton = screen.getByLabelText("Next Testimonial")
    fireEvent.click(nextButton)
    
    // Should show Alibaba testimonial
    expect(screen.getByText("Sanhong Li")).toBeTruthy()
    expect(screen.getByText(/As a member of the Adoptium Working Group, Alibaba Cloud/)).toBeTruthy()
  })

  it("navigates to previous testimonial when previous button is clicked", () => {
    render(<Testimonials />)
    
    // Click previous button (should wrap to last testimonial)
    const prevButton = screen.getByLabelText("Previous Testimonial")
    fireEvent.click(prevButton)
    
    // Should show Azul testimonial (last one)
    expect(screen.getByText("Simon Ritter")).toBeTruthy()
    expect(screen.getByText("Deputy CTO at Azul Systems")).toBeTruthy()
  })

  it("navigates using progress indicators", () => {
    render(<Testimonials />)
    
    // Initial state - first testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Click on third progress indicator
    const thirdIndicator = screen.getByLabelText("Go to testimonial 3")
    fireEvent.click(thirdIndicator)
    
    // Should show Azul testimonial
    expect(screen.getByText("Simon Ritter")).toBeTruthy()
    expect(screen.getByText(/Azul has been a member of the Eclipse Adoptium/)).toBeTruthy()
  })

  it("navigates using company logos", () => {
    render(<Testimonials />)
    
    // Initial state - Microsoft testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Click on Alibaba logo
    const alibabaButton = screen.getByLabelText("Alibaba Cloud Testimonial")
    fireEvent.click(alibabaButton)
    
    // Should show Alibaba testimonial
    expect(screen.getByText("Sanhong Li")).toBeTruthy()
    expect(screen.getByText("Director of Compiler & Runtime, Alibaba Cloud Intelligence")).toBeTruthy()
  })

  it("auto-advances testimonials after 8 seconds", async () => {
    render(<Testimonials />)
    
    // Initial state - Microsoft testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Fast-forward 8 seconds
    act(() => {
      vi.advanceTimersByTime(8000)
    })
    
    // Should now show Alibaba testimonial
    await waitFor(() => {
      expect(screen.getByText("Sanhong Li")).toBeTruthy()
    })
  })

  it("cycles through all testimonials automatically", async () => {
    render(<Testimonials />)
    
    // Start with Microsoft
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // After 8 seconds -> Alibaba
    act(() => {
      vi.advanceTimersByTime(8000)
    })
    await waitFor(() => {
      expect(screen.getByText("Sanhong Li")).toBeTruthy()
    })
    
    // After another 8 seconds -> Azul
    act(() => {
      vi.advanceTimersByTime(8000)
    })
    await waitFor(() => {
      expect(screen.getByText("Simon Ritter")).toBeTruthy()
    })
    
    // After another 8 seconds -> back to Microsoft
    act(() => {
      vi.advanceTimersByTime(8000)
    })
    await waitFor(() => {
      expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    })
  })

  it("resets auto-advance timer when manually navigating", async () => {
    render(<Testimonials />)
    
    // Initial state - Microsoft testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Fast-forward 4 seconds (halfway to auto-advance)
    act(() => {
      vi.advanceTimersByTime(4000)
    })
    
    // Manually navigate to next
    const nextButton = screen.getByLabelText("Next Testimonial")
    fireEvent.click(nextButton)
    
    // Should show Alibaba testimonial
    expect(screen.getByText("Sanhong Li")).toBeTruthy()
    
    // Fast-forward another 4 seconds (should not auto-advance yet since timer was reset)
    act(() => {
      vi.advanceTimersByTime(4000)
    })
    
    // Should still be on Alibaba
    expect(screen.getByText("Sanhong Li")).toBeTruthy()
    
    // Fast-forward another 4 seconds to complete the 8-second cycle
    act(() => {
      vi.advanceTimersByTime(4000)
    })
    
    // Now should auto-advance to Azul
    await waitFor(() => {
      expect(screen.getByText("Simon Ritter")).toBeTruthy()
    })
  })

  it("handles clicking on the same progress indicator", () => {
    render(<Testimonials />)
    
    // Initial state - first testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
    
    // Click on first progress indicator (current one)
    const firstIndicator = screen.getByLabelText("Go to testimonial 1")
    fireEvent.click(firstIndicator)
    
    // Should remain on Microsoft testimonial
    expect(screen.getByText("Martijn Verburg")).toBeTruthy()
  })

  it("has proper accessibility attributes", () => {
    render(<Testimonials />)
    
    // Check for proper ARIA labels
    expect(screen.getByLabelText("Previous Testimonial")).toBeTruthy()
    expect(screen.getByLabelText("Next Testimonial")).toBeTruthy()
    expect(screen.getByLabelText("Microsoft Testimonial")).toBeTruthy()
    expect(screen.getByLabelText("Alibaba Cloud Testimonial")).toBeTruthy()
    expect(screen.getByLabelText("Azul Systems Testimonial")).toBeTruthy()
    
    // Check for proper alt text on images
    expect(screen.getByAltText("Martijn Verburg")).toBeTruthy()
    expect(screen.getByAltText("Microsoft")).toBeTruthy()
    expect(screen.getByAltText("Alibaba Cloud")).toBeTruthy()
    expect(screen.getByAltText("Azul Systems")).toBeTruthy()
  })

  it("renders all testimonial data correctly", () => {
    render(<Testimonials />)
    
    const testimonials = [
      {
        name: "Martijn Verburg",
        role: "Principal Group Manager - Java & Go @ Microsoft",
        company: "Microsoft"
      },
      {
        name: "Sanhong Li", 
        role: "Director of Compiler & Runtime, Alibaba Cloud Intelligence",
        company: "Alibaba Cloud"
      },
      {
        name: "Simon Ritter",
        role: "Deputy CTO at Azul Systems", 
        company: "Azul Systems"
      }
    ]
    
    testimonials.forEach((testimonial) => {
      // Navigate to each testimonial
      const companyButton = screen.getByLabelText(`${testimonial.company} Testimonial`)
      fireEvent.click(companyButton)
      
      // Check if testimonial data is displayed
      expect(screen.getByText(testimonial.name)).toBeTruthy()
      expect(screen.getByText(testimonial.role)).toBeTruthy()
      expect(screen.getByAltText(testimonial.name)).toBeTruthy()
      expect(screen.getByAltText(testimonial.company)).toBeTruthy()
    })
  })

  it("displays correct testimonial structure", () => {
    render(<Testimonials />)
    
    // Check that the main elements exist
    expect(screen.getByRole("blockquote")).toBeTruthy()
    expect(screen.getByRole("figure")).toBeTruthy()
    
    // Check that images have proper styling containers
    const profileImage = screen.getByAltText("Martijn Verburg")
    expect(profileImage.className).toContain("rounded-full")
    
    // Check that company logos exist
    const microsoftLogo = screen.getByAltText("Microsoft")
    expect(microsoftLogo).toBeTruthy()
  })
})
