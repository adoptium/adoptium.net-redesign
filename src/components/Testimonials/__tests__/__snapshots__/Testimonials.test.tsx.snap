// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testimonials component > should render correctly 1`] = `
<div>
  <section
    class="relative bg-gradient-to-br from-[#0E002A] via-[#1A0B3D] to-[#0E002A] overflow-hidden"
  >
    <div
      class="absolute inset-0 opacity-5"
    >
      <div
        class="absolute inset-0 bg-repeat"
      />
    </div>
    <div
      class="relative max-w-7xl px-6 py-16 mx-auto text-center lg:py-24"
    >
      <div
        class="relative max-w-5xl mx-auto"
      >
        <div
          class="bg-white/5 backdrop-blur-sm rounded-3xl border border-white/10 p-8 md:p-12 shadow-2xl"
        >
          <figure>
            <div
              class="mb-8"
            >
              <svg
                class="h-12 w-12 mx-auto text-pink opacity-60"
                fill="none"
                viewBox="0 0 24 27"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.017 18L14.017 10.609C14.017 4.905 17.748 1.039 23 0L23.995 2.151C21.563 3.068 20 5.789 20 8H24V18H14.017ZM0 18V10.609C0 4.905 3.748 1.038 9 0L9.996 2.151C7.563 3.068 6 5.789 6 8H9.983L9.983 18L0 18Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <blockquote
              class="slide-in-right mb-8"
            >
              <div
                class="min-h-[200px] md:min-h-[180px] flex items-center justify-center"
              >
                <p
                  class="text-xl md:text-2xl leading-relaxed font-medium text-white/90 italic"
                >
                  "
                  Microsoft is proud to be a Strategic Member of the Eclipse Adoptium Working Group, collaborating closely with industry partners to contribute to and benefit from a vibrant open-source ecosystem. Together, we are dedicated to delivering high-quality, secure, and performant Java runtimes that empower our customers and the developer community.
                  "
                </p>
              </div>
            </blockquote>
            <figcaption
              class="flex flex-col sm:flex-row items-center justify-center gap-4"
            >
              <img
                alt="Martijn Verburg"
                class="w-16 h-16 rounded-full border-2 border-pink-400/30 object-cover"
                src="/images/authors/martijnverburg.jpg"
              />
              <div
                class="text-center sm:text-left"
              >
                <div
                  class="text-lg font-semibold text-white"
                >
                  Martijn Verburg
                </div>
                <div
                  class="text-pink font-medium"
                >
                  Principal Group Manager - Java & Go @ Microsoft
                </div>
              </div>
            </figcaption>
          </figure>
        </div>
      </div>
      <div
        class="flex justify-center items-center mt-12 space-x-6"
      >
        <button
          aria-label="Previous Testimonial"
          class="group p-3 rounded-full bg-white/10 border border-white/20 hover:bg-white/20 hover:border-pink-400/50 transition-all duration-300"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-pink transition-colors"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            />
          </svg>
        </button>
        <div
          class="flex space-x-2"
        >
          <button
            aria-label="Go to testimonial 1"
            class="h-2 rounded-full transition-all duration-300 w-8 bg-pink-400"
          />
          <button
            aria-label="Go to testimonial 2"
            class="h-2 rounded-full transition-all duration-300 w-2 bg-white/30 hover:bg-white/50"
          />
          <button
            aria-label="Go to testimonial 3"
            class="h-2 rounded-full transition-all duration-300 w-2 bg-white/30 hover:bg-white/50"
          />
        </div>
        <button
          aria-label="Next Testimonial"
          class="group relative p-3 rounded-full bg-white/10 border border-white/20 hover:bg-white/20 hover:border-pink-400/50 transition-all duration-300 overflow-hidden"
        >
          <svg
            class="w-5 h-5 text-white group-hover:text-pink transition-colors relative z-10"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
            />
          </svg>
          <div
            class="absolute inset-0 rounded-full border-2 border-transparent border-t-pink-400 animate-spin opacity-60"
            style="animation-duration: 8s;"
          />
        </button>
      </div>
      <div
        class="flex justify-center items-center mt-12 space-x-8"
      >
        <button
          aria-label="Microsoft Testimonial"
          class="transition-all duration-300 opacity-60 hover:opacity-100 opacity-100 scale-110"
        >
          <div
            class="h-12 w-32 flex items-center justify-center"
          >
            <img
              alt="Microsoft"
              class="max-h-full max-w-full object-contain brightness-0 invert"
              src="/images/microsoft.svg"
            />
          </div>
        </button>
        <button
          aria-label="Alibaba Cloud Testimonial"
          class="transition-all duration-300 opacity-60 hover:opacity-100 hover:scale-105"
        >
          <div
            class="h-12 w-32 flex items-center justify-center"
          >
            <img
              alt="Alibaba Cloud"
              class="max-h-full max-w-full object-contain brightness-0 invert"
              src="/images/alibaba.png"
            />
          </div>
        </button>
        <button
          aria-label="Azul Systems Testimonial"
          class="transition-all duration-300 opacity-60 hover:opacity-100 hover:scale-105"
        >
          <div
            class="h-12 w-32 flex items-center justify-center"
          >
            <img
              alt="Azul Systems"
              class="max-h-full max-w-full object-contain brightness-0 invert"
              src="/images/azul.svg"
            />
          </div>
        </button>
      </div>
    </div>
  </section>
</div>
`;
